test/EFCore.PG.FunctionalTests/Query/SpatialQueryNpgsqlGeographyTest.cs

```diff
@@ -0,0 +1,201 @@
+        public override Task EndPoint(bool isAsync)                  => Task.CompletedTask;
+        public override Task Envelope(bool isAsync)                  => Task.CompletedTask;
+        public override Task EqualsTopologically(bool isAsync)       => Task.CompletedTask;
+        public override Task ExteriorRing(bool isAsync)              => Task.CompletedTask;
+        public override Task GetGeometryN(bool isAsync)              => Task.CompletedTask;
+        public override Task GetInteriorRingN(bool isAsync)          => Task.CompletedTask;
+        public override Task GetPointN(bool isAsync)                 => Task.CompletedTask;
+        public override Task ICurve_IsClosed(bool isAsync)           => Task.CompletedTask;
+        public override Task IGeometryCollection_Count(bool isAsync) => Task.CompletedTask;
+        public override Task IMultiCurve_IsClosed(bool isAsync)      => Task.CompletedTask;
+        public override Task IsEmpty(bool isAsync)                   => Task.CompletedTask;
+        public override Task IsRing(bool isAsync)                    => Task.CompletedTask;
+        public override Task IsSimple(bool isAsync)                  => Task.CompletedTask;
+        public override Task IsValid(bool isAsync)                   => Task.CompletedTask;
+        public override Task Item(bool isAsync)                      => Task.CompletedTask;
+        public override Task LineString_Count(bool isAsync)          => Task.CompletedTask;
+        public override Task M(bool isAsync)                         => Task.CompletedTask;
+        public override Task NumGeometries(bool isAsync)             => Task.CompletedTask;
+        public override Task NumInteriorRings(bool isAsync)          => Task.CompletedTask;
+        public override Task NumPoints(bool isAsync)                 => Task.CompletedTask;
+        public override Task Overlaps(bool isAsync)                  => Task.CompletedTask;
+        public override Task PointOnSurface(bool isAsync)            => Task.CompletedTask;
+        public override Task Relate(bool isAsync)                    => Task.CompletedTask;
+        public override Task Reverse(bool isAsync)                   => Task.CompletedTask;
+        public override Task StartPoint(bool isAsync)                => Task.CompletedTask;
+        public override Task SymmetricDifference(bool isAsync)       => Task.CompletedTask;
+        public override Task Touches(bool isAsync)                   => Task.CompletedTask;
+        public override Task Union(bool isAsync)                     => Task.CompletedTask;
+        public override Task Union_void(bool isAsync)                => Task.CompletedTask;
+        public override Task Within(bool isAsync)                    => Task.CompletedTask;
+        public override Task X(bool isAsync)                         => Task.CompletedTask;
+        public override Task Y(bool isAsync)                         => Task.CompletedTask;
```

Comment:
X and Y aren't supported? (We translate to `Long` and `Lat`)

Reply:
I wondered about this too, but there really doesn't seem to be a way... [Here's the PostGIS supported function matrix](https://postgis.net/docs/manual-2.5/PostGIS_Special_Functions_Index.html#PostGIS_TypeFunctionMatrix), `ST_X/Y` are unsupported for geography and no long/lat alternative seems to be available. [Various](https://gis.stackexchange.com/questions/229086/get-latitude-and-longitude-of-a-geography-column/229215) [stackoverflow](https://gis.stackexchange.com/questions/57018/get-lon-lat-values-from-a-geography-data-type-in-postgis) questions seem to show people stuck on this.

In general geography seems less usable on PG than it might be on SqlServer (in terms of the functions supported). That's also why I think geometry is a more sane default for Npgsql than geography.